app/src/main/java/com/physical_web/cms/physicalwebcms/SetupManager.java

```diff
@@ -0,0 +1,48 @@
+package com.physical_web.cms.physicalwebcms;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.preference.PreferenceManager;
+
+/**
+ * This class ensures all auxiliary conditions required to run the app are met. Such conditions
+ * include network connectivity and Google Drive authourization. It will guide
+ * the user through the setup of the app the first time they run it. It will also constantly check
+ * if the conditions for running the app are met after that.
+ */
+
+public class SetupManager {
+    SharedPreferences sharedPreferences;
+
+    public SetupManager(Context context) {
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+    }
+
+    public void checkRequirements(Context context) {
+        if(isFirstRun()) {
+            Intent welcomeIntent = new Intent(context, WelcomeActivity.class);
+            context.startActivity(welcomeIntent);
```

Comment:
If I run the code...  crash the app.

![screen shot 2017-06-19 at 10 32 01 am](https://user-images.githubusercontent.com/1067352/27278874-971d9fe8-54da-11e7-9544-b87fcf7a3b6c.png)

I'm not quite sure, but if you use `welcomeIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);` before start the new activity could fix the problem.